### PR TITLE
Add a vim style command to change the tab size

### PR DESCRIPTION
**Pull request by YouTwitFace**

### CODE REVIEW

File: babi/horizontal_scrolling.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def scrolled_line(s: str, x: int, width: int) -> str:
 class _CalcWidth:
     @cached_property
     def _window(self) -> 'curses._CursesWindow':
-        return curses.newwin(1, 10)
+        return curses.newwin(10, 10)
```
Comment (asottile) [inline babi/horizontal_scrolling.py:38]: if this is overflowing the first line requiring more lines then there's a bug

File: babi/file.py [hunk 2]
```diff
@@ -201,11 +201,13 @@ def __init__(
             initial_line: int,
             color_manager: ColorManager,
             hl_factories: Tuple[HLFactory, ...],
+            tab_size: int = 4,
     ) -> None:
         self.filename = filename
         self.initial_line = initial_line
         self.modified = False
-        self.buf = Buf([])
+        self.tab_size = tab_size
```
Comment (asottile) [inline babi/file.py:209]: ideally we don't store this state in two places -- can it be in `buf` only?

File: babi/main.py [hunk 3]
```diff
@@ -124,6 +125,9 @@ def _filenames(filenames: List[str]) -> Tuple[List[Optional[str]], List[int]]:
 def main(argv: Optional[Sequence[str]] = None) -> int:
     parser = argparse.ArgumentParser()
     parser.add_argument('filenames', metavar='filename', nargs='*')
+    parser.add_argument(
+        '-T, --tabsize', default=4, type=int, metavar='cols', dest='tab_size',
+    )
```
Comment (asottile) [inline babi/main.py:130]: let's not add an option, I'd like to do this differently later and not have to break compat for this

File: babi/horizontal_scrolling.py (hunk 1 above)
Comment (YouTwitFace) [reply babi/horizontal_scrolling.py:38]: It was either this or actual tabs wouldn't re-render when changing the tab size.

Comment (asottile) [reply babi/horizontal_scrolling.py:38]: the height should not change, have you tried changing the width?

Comment (YouTwitFace) [reply babi/horizontal_scrolling.py:38]: That still fails.

**Comment by itshaydendev** [comment]
Maybe this could have a feature where (like Vim) you can get a selection of text and hit `<` or `>` to move the indent accordingly?

**Comment by YouTwitFace** [comment]
@itshaydendev you can already do that.

**Pull request merged**